### PR TITLE
Create dummy classes and test cases to start CI

### PR DESCRIPTION
**Pull request by I-Bouros**
Created dummy classes and test cases to start implementing CI and Git Action workflows.

Takes into account design issues as raised in #7 , #2, #8 and #9 .

Uses as inspiration for the set up the following repository files: https://github.com/I-Bouros/2020-software-engineering-projects-pk/tree/master/pkmodel

### CODE REVIEW

**Review by DavAug — CHANGES_REQUESTED** [review]
Thanks @I-Bouros those base classes and tests look really good. 

Would it be possible to separate those classes from setting up the GitHub actions? The GitHub actions should ideally just be set up with a simple dummy function which will be removed with the first functional code merge.

**Comment by I-Bouros** [comment]
> Thanks @I-Bouros those base classes and tests look really good.
> 
> Would it be possible to separate those classes from setting up the GitHub actions? The GitHub actions should ideally just be set up with a simple dummy function which will be removed with the first functional code merge.

Do you mean set up a different file dummy.py with its pair test_dummy.py? And these are to be deleted once the Git Actions are set up?

**Comment by DavAug** [comment]
> > Thanks @I-Bouros those base classes and tests look really good.
> 
> > 
> 
> > Would it be possible to separate those classes from setting up the GitHub actions? The GitHub actions should ideally just be set up with a simple dummy function which will be removed with the first functional code merge.
> 
> 
> 
> Do you mean set up a different file dummy.py with its pair test_dummy.py? And these are to be deleted once the Git Actions are set up?

Yes exactly, that's a good idea.

And please remove the model, simulation controller classes etc from this PR for now, I.e. separate them out in different branches so we can address the issues one by one.

**Comment by I-Bouros** [comment]
> > > Thanks @I-Bouros those base classes and tests look really good.
> > 
> > 
> > > 
> > 
> > 
> > > Would it be possible to separate those classes from setting up the GitHub actions? The GitHub actions should ideally just be set up with a simple dummy function which will be removed with the first functional code merge.
> > 
> > 
> > Do you mean set up a different file dummy.py with its pair test_dummy.py? And these are to be deleted once the Git Actions are set up?
> 
> Yes exactly, that's a good idea.
> 
> And please remove the model, simulation controller classes etc from this PR for now, I.e. separate them out in different branches so we can address the issues one by one.

I have separated them as of now in distinct branches; pulled requests for each one of them #12 , #14, #13 and of course #10.

Have invited everyone to review them.

File: branchpro/__init__.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"""branchpro is a Branching Processes modelling library.
+It contains functionality for modelling, simulating, and visualising the
+number of cases of infections by day during an outbreak of the influenza virus.
```
Comment (rcw5890) [inline branchpro/__init__.py:3]: Maybe we will need a copyright or license mention here as well? @DavAug , is this something you thought about?

Also, on their init file Pints has `# Root of the pints module. # Provides access to all shared functionality (optimisation, mcmc, etc.).` Maybe we should add another comment similar to this.

File: branchpro/__init__.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+"""
+
+# Import version info
+from .version_info import VERSION_INT, VERSION  # noqa
+
+# Import main classes
+from .models import ForwardModel, BranchProModel    # noqa
```
Comment (rcw5890) [inline branchpro/__init__.py:10]: I think these lines will fail until we get the next pull requests in. I would remove them for now, and add them back in within those pull requests.

File: branchpro/__init__.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+from .simulation import SimulationController    # noqa
+from .visualisation import Visualisation     # noqa
```
Comment (rcw5890) [inline branchpro/__init__.py:12]: There's a warning about no newline. I think it's referring to [this requirement](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline).

**Comment by rcw5890** [comment]
This is a very strong start on this project, thanks for your hard work @I-Bouros ! 

I read through the dummy test and made a few minor suggestions.

On the topic of testing, we may want to move to a "run-tests.py" format along the lines of https://github.com/pints-team/pints/blob/master/run-tests.py . I'm happy to defer that until after we get Github actions running.

File: branchpro/__init__.py (hunk 1 above)
Comment (I-Bouros) [reply branchpro/__init__.py:11]: Just added with the last commit!

File: branchpro/__init__.py (hunk 3 above)
Comment (I-Bouros) [reply branchpro/__init__.py]: Noted; made the required change

File: branchpro/__init__.py (hunk 2 above)
Comment (I-Bouros) [reply branchpro/__init__.py]: Noted; made the required change

**Comment by I-Bouros** [comment]
> This is a very strong start on this project, thanks for your hard work @I-Bouros !
> 
> I read through the dummy test and made a few minor suggestions.
> 
> On the topic of testing, we may want to move to a "run-tests.py" format along the lines of https://github.com/pints-team/pints/blob/master/run-tests.py . I'm happy to defer that until after we get Github actions running.

I have completed the suggested changes, as it cane seen in the last couple of commits; also changed the other branches submitted for pull to illustrate these points. We can definitely do something similar to the pints' "run-tests.py" once the CI is set up, but I tried for now to follow @DavAug point of not making it too complicated at this stage.

**Review by rcw5890 — APPROVED** [review]
Thanks, looks ready to go in to me. 

@DavAug , I saw you requested changes earlier, did you want to do a review on this one now?

**Pull request merged**